bindings/rust/bench/benches/throughput.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use bench::{
+    CipherSuite::{self, *},
+    CryptoConfig,
+    ECGroup::X25519,
+    OpenSslHarness, RustlsHarness, S2NHarness, TlsBenchHarness,
+};
+use criterion::{
+    black_box, criterion_group, criterion_main, measurement::WallTime, BatchSize, BenchmarkGroup,
+    Criterion, Throughput,
+};
+use std::any::type_name;
+
+pub fn bench_throughput_cipher_suite(c: &mut Criterion) {
+    let mut shared_buf = [0u8; 100000];
+
+    fn bench_throughput_for_library<T: TlsBenchHarness>(
+        bench_group: &mut BenchmarkGroup<WallTime>,
+        shared_buf: &mut [u8],
+        cipher_suite: &CipherSuite,
+    ) {
+        bench_group.bench_function(type_name::<T>(), |b| {
+            b.iter_batched_ref(
+                || {
+                    let mut harness = T::new(&CryptoConfig {
+                        cipher_suite: *cipher_suite,
+                        ec_group: X25519,
+                    })
+                    .unwrap();
+                    harness.handshake().unwrap();
+                    harness
+                },
+                |harness| harness.round_trip_transfer(black_box(shared_buf)).unwrap(),
```

Comment:
What is this black_box function doing?

Reply:
It's there to keep the compiler from optimizing the function away.